File: opensource/test/src/test/java/com/sos/scheduler/engine/tests/jira/js644/continuous/TestJob.java
package com.sos.scheduler.engine.tests.jira.js644.continuous;

import sos.spooler.Job_impl;

import static java.lang.Thread.sleep;

public class TestJob extends Job_impl {
    @Override public final boolean spooler_open() throws InterruptedException {
        sleep(5000);
        return true;
    }

    @Override public final boolean spooler_process() throws InterruptedException {
        sleep(1000);
        return true;
    }
}

File: engine-tests/src/test/scala/com/sos/scheduler/engine/tests/jira/js1301/TestJob.scala
package com.sos.scheduler.engine.tests.jira.js1301

import java.lang.Thread.sleep

/**
 * @author Joacim Zschimmer
 */
final class TestJob extends sos.spooler.Job_impl {

  override def spooler_process() = {
    spooler_log.info(s"API TEST_AGENT=/${sys.env("TEST_AGENT")}/")
    spooler_task.order.params.value("sleep") match {
      case "" ⇒
      case seconds ⇒ sleep(seconds.toInt * 1000)
    }
    true
  }
}
